common/strings.c

```diff
@@ -88,6 +88,57 @@ TDNFSafeAllocateString(
     goto cleanup;
 }
 
+uint32_t
```

Comment:
if split into a count and collect, can avoid realloc

Reply:
Here is my justification for using `realloc()` in this instance. I'm open for suggestions/corrections:

- The number of realloc() calls are logarithmic to the target size and we start off with a large chunk (16 in this case) and double each time.
- `strtok()` (or other tokenizing methods `strsep()` etc) cobblers the input string so I'd have to make two copies (`strdup()`-ed) to achieve this effect. This makes the code look ugly :).
- The talingling resize realloc() does no copy or dealloc most of the time.
- This progressive growth pattern is easy understand and looks good

Reply:
you dont have to use strtok. you have some separator - space in this case. just count space instances using an advancing ptr and strchr. i think its much cleaner to know what you need to do and then allocate precisely for that.
again, just my opinion.

Reply:
Switched to count-and-alloc